azure-pipelines-official.yml

```diff
@@ -135,27 +136,51 @@ extends:
         displayName: Official Build
         timeoutInMinutes: 360
         templateContext:
+          # This ensures that the SDL analysis runs only once for this whole folder and not for each output separately.
+          # https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/features/outputs#multiple-outputs
+          # Furthermore, our uploaded outputs contain only nupkg/zip files which some binary analyzers like BinSkim would ignore,
+          # but analyzing the whole artifacts directory makes these analyzers scan the DLLs as well.
+          outputParentDirectory: '$(Build.SourcesDirectory)/artifacts'
+
           outputs:
 
+          - output: pipelineArtifact
+            displayName: Publish SBOM manifest
+            continueOnError: true
+            targetPath: '$(Build.SourcesDirectory)/artifacts/sbom'
+            artifactName: $(ARTIFACT_NAME)
+
+          # Publish OptProf configuration files to the artifact service
+          # This uses the ArtifactServices.Drop.PAT build variable which is required to enable cross account access using PAT (dnceng -> devdiv)
+          - output: artifactsDrop
+            displayName: 'OptProf - Publish to Artifact Services - ProfilingInputs'
+            condition: succeeded()
+            dropServiceURI: 'https://devdiv.artifacts.visualstudio.com'
+            buildNumber: 'ProfilingInputs/$(System.TeamProject)/$(Build.Repository.Name)/$(SourceBranchName)/$(Build.BuildNumber)'
+            sourcePath: '$(Build.SourcesDirectory)/artifacts/OptProf/$(BuildConfiguration)/Data'
+            toLowerCase: false
+            usePat: false
+            retentionDays: 90
+
           # Publish OptProf generated JSON files as a pipeline artifact. This allows for easy inspection from
           # a build execution.
           - output: pipelineArtifact
             displayName: 'Publish OptProf Data Files'
             condition: succeeded()
-            targetPath: '$(Build.SourcesDirectory)\artifacts\OptProf\$(BuildConfiguration)\Data'
+            targetPath: '$(Build.SourcesDirectory)/artifacts\OptProf\$(BuildConfiguration)\Data'
             artifactName: 'OptProf Data Files'
 
           - output: pipelineArtifact
             displayName: 'Publish Logs'
             condition: succeededOrFailed()
-            targetPath: '$(Build.SourcesDirectory)\artifacts\log\$(BuildConfiguration)'
+            targetPath: '$(Build.SourcesDirectory)/artifacts\log\$(BuildConfiguration)'
```

Comment:
Why did you switch the directory separator char in only one place?

Reply:
The `outputParentDirectory: '$(Build.SourcesDirectory)/artifacts'` prefix is compared as string with each target path, so the directory separator needs to match there. I can update them all to match though.

Reply:
Where does that comparison happen?

Reply:
https://dev.azure.com/dnceng/1ESPipelineTemplates/_git/1ESPipelineTemplates?path=/v1/Steps/SDLBinaryAnalysis.yml&version=GBmain&line=65&lineEnd=66&lineStartColumn=1&lineEndColumn=132&lineStyle=plain&_a=contents